### DISCUSSION TITLE
Forward chat messages

### DISCUSSION THREAD

**Discussion by Moshyfawn** [zed-industries/zed#23915] [Feature Requests]
## Prelude 

This is more of a community management type request, but as Zed is more than an IDE and strives to be a collaboration platform, there's a fair chance there will be more use cases where communities would like to see moderation tools and better chat management support UX.

## Description

I'd like to be able to forward chat messages as replies. There are quite a few people in the #zed channels _(and others)_ who are looking for support on how to use certain Zed features, some of which are repeat questions that I or another community member has already answered. Currently, I have to scrub through the chat history manually _(another request #10139)_ to find the answer I have already given, copy it using the `Copy Message Text` action, and paste it into my reply. It would be nice to be able to forward the message instead.

I haven't thought the whole UX through yet, as there are quite a few edge cases where I'm not sure how the feature might work. My initial idea is that, in addition to `Copy message text` and `Delete message`, there will be another action called `Forward message` which will put the message into the textarea, where you can then click the reply button on the message you want to forward to and send the message.

The forwarded message can be visually distinguished by some sort of border or just a label next to where says "edited" for an edited message.

I'd also like to be able to click on the forwarded message to scroll to / highlight the original forwarded message.

These types of "management" actions can be put behind a collaboration role like `Admin` / `Member` etc.

**Comment by maxdeviant** [comment]
We can potentially explore this when we revisit the chat functionality. Going to convert to a discussion, for now.
